### PR TITLE
Adapt Discovery to make it compatible with NodeJs env

### PR DESCRIPTION
**Pull request by dmitriyslaym**
#84

### CODE REVIEW

File: packages/scalecube-microservice/src/Microservices/Microservices.ts [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
+import { isNodeEnv } from '../helpers/utils'
 import { createServiceRegistry } from '../Registry/ServiceRegistry';
 import { createMethodRegistry } from '../Registry/MethodRegistry';
 import { MicroserviceContext } from '../helpers/types';
 import { Endpoint, Message, Microservice, MicroserviceOptions, Microservices as MicroservicesInterface } from '../api';
 import { ASYNC_MODEL_TYPES, MICROSERVICE_NOT_EXISTS } from '../helpers/constants';
 
 export const Microservices: MicroservicesInterface = Object.freeze({
-  create: ({ services, seedAddress = location.hostname }: MicroserviceOptions): Microservice => {
+  create: ({ services, seedAddress = isNodeEnv() ? 'clusterAddress' : location.hostname }: MicroserviceOptions): Microservice => {
```
Comment (idanilt) [inline packages/scalecube-microservice/src/Microservices/Microservices.ts]: location.hostname, seem pointless in here, "defaultSeedAddress" will work much better

File: packages/scalecube-discovery/src/Discovery/DiscoveryActions.ts [hunk 2]
```diff
@@ -7,11 +7,13 @@ import {
   LeaveCluster,
   ScalecubeGlobal,
 } from '../helpers/types';
+import { getGlobal } from '../helpers/utils'
 
 export const getCluster = ({ seedAddress }: GetCluster): Cluster => {
-  window.scalecube = window.scalecube || ({} as ScalecubeGlobal);
-  window.scalecube.clusters = window.scalecube.clusters || ({} as ClustersMap);
-  const namespace = window.scalecube.clusters;
+  const globalNamespace = getGlobal()
```
Comment (idanilt) [inline packages/scalecube-discovery/src/Discovery/DiscoveryActions.ts]: getGlobalScope => globalNamespace.scalecube will look better
const scalecubeGlobal = getGlobalScope()
scalecubeGlobal.clusters

File: packages/scalecube-discovery/src/helpers/utils.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export const isNodeEnv = () => typeof window === 'undefined';
```
Comment (idanilt) [inline packages/scalecube-discovery/src/helpers/utils.ts]: `isNodeEnv` function doesn't check for NodeJS, it's checking for `window`
actually, it seem obsolete

File: packages/scalecube-discovery/src/helpers/utils.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
...
+
+export const getGlobal = () => isNodeEnv() ? global : window;
```
Comment (idanilt) [inline packages/scalecube-discovery/src/helpers/utils.ts]: see my earlier comment 
I would implement it: window ==> global ==> WorkerGlobalScope ==> {}

File: packages/scalecube-microservice/src/helpers/utils.ts [hunk 5]
```diff
@@ -18,10 +18,4 @@ export const throwErrorFromServiceCall = ({
...
 };
 
-/* tslint:disable */
-// https://stackoverflow.com/questions/105034/create-guid-uuid-in-javascript
-export const uuidv4 = () =>
-  ('' + 1e7 + -1e3 + -4e3 + -8e3 + -1e11).replace(/[018]/g, (c: any) =>
-    (c ^ (crypto.getRandomValues(new Uint8Array(1))[0] & (15 >> (c / 4)))).toString(16)
-  );
-/* tslint:enable */
+export const isNodeEnv = () => typeof window === 'undefined';
```
Comment (idanilt) [inline packages/scalecube-microservice/src/helpers/utils.ts]: seem obsolete and duplication

Comment (dmitriyslaym) [reply packages/scalecube-microservice/src/helpers/utils.ts]: @idanilt This is a helper in a different package, I thought some helpers, if they are required in different packages

Comment (dmitriyslaym) [reply packages/scalecube-microservice/src/helpers/utils.ts]: @idanilt Or are you suggesting to create a package for common helpers?

**Pull request merged**